lib/src/bloc/movies_bloc.dart

```diff
@@ -1,60 +1,107 @@
 import 'dart:async';
-import '../model/movies.dart';
+import '../events.dart';
 import '../resources/movies_repository.dart';
+import '../state_type.dart';
 import 'i_movies_bloc.dart';
 
 class MoviesBloc extends IMoviesBloc {
   bool isTextFieldEmpty = true;
   late MoviesRepository _moviesRepository;
-  StreamController<Movies> _trendingMoviesStream =
-      StreamController<Movies>.broadcast();
-  StreamController<Movies> _discoverMoviesStream =
-      StreamController<Movies>.broadcast();
+
+  StreamController<StateType> _trendingMoviesStream =
+      StreamController<StateType>.broadcast();
+
+  StreamController<StateType> _discoverMoviesStream =
+      StreamController<StateType>.broadcast();
 
   MoviesBloc([MoviesRepository? moviesRepository])
       : _moviesRepository = moviesRepository ?? MoviesRepository();
 
   @override
-  void initialize() {}
+  Stream<StateType> get cardSwiperMoviesStream => _trendingMoviesStream.stream;
 
   @override
-  void dispose() {
-    _trendingMoviesStream.close();
-    _discoverMoviesStream.close();
-  }
+  Stream<StateType> get gridMoviesStream => _discoverMoviesStream.stream;
 
-  void fetchTrendingMovies() async {
-    final trendingMovieList = await _moviesRepository.fetchTrendingMovies();
-    _trendingMoviesStream.sink.add(
-      trendingMovieList,
-    );
-  }
-
-  void fetchDiscoverMovies() async {
-    final discoverMovieList = await _moviesRepository.fetchDiscoverMovies();
-    _discoverMoviesStream.sink.add(
-      discoverMovieList,
-    );
+  void getMoviesState(Events event, [String movieName = ""]) async {
```

Comment:
Here i think you have some kind of similar functionality that can be abstracted to a function, are you agree with that?

Reply:
done